### PR TITLE
(1022) New Python + Ruby ingest implementation

### PR DESCRIPTION
**Pull request by leeky**
Replacement for the lambda-based ingest, which isn't available on GPaaS.

We use Python's 'csvkit' library to parse the Excel files into CSV files (one per sheet). Ruby then reads these files, parses, validates and finally loads them into the database.

There's a fairly comprehensive test suite, and I've collected a number of edge cases that will get full end-to-end tests in a forthcoming PR.

NB: I've not yet implemented this as a Sidekiq worker. When I do this, I'll add in the error handling, so that Sidekiq won't endless retry jobs that will *never* pass - e.g. when the Excel file is unreadable.

### CODE REVIEW

**Comment by rgarner** [comment]
@leeky without going through it all right now, would it be possible to ask for some class doc comments at the tops of things like `Compare`? That would be hugely helpful in review. Thanks

File: Dockerfile [hunk 1]
```diff
@@ -15,6 +15,8 @@ RUN YARN_VERSION=1.9.4 \
   && ln -s /opt/yarn/bin/yarn /usr/local/bin/yarnpkg \
   && rm yarn-v$YARN_VERSION.tar.gz
 
+RUN apt-get install -qq -y python python-pip python-setuptools python-wheel --fix-missing --no-install-recommends && pip install --quiet csvkit
```
Comment (rjw1) [inline Dockerfile]: We should probably use a requirements.txt here so that the PaaS can do magic later.

File: app/models/ingest/converter.rb [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+    def fetch_row_count(file)
+      command = "wc -l < #{file} | xargs"
+      row_count = Ingest::CommandRunner.new(command).run!.stdout.first.to_i
+      row_count -= 1 unless row_count.zero? # Handle empty results
+      row_count
+    end
+
+    def invoice_sheet_name
+      @invoice_sheet_name ||= sheets.find { |sheet| sheet.match(/(booking|finance|management|invoice)/i) }
```
Comment (james) [inline app/models/ingest/converter.rb]: Interesting! So we're changing our explicit sheet name whitelisting to a regex match? This is probably better assuming you've checked all the existing sheet names.
